src/cswinrt/code_writers.h

```diff
@@ -4065,18 +3810,54 @@ R"(file static class %InterfaceEntriesImpl
 )", name, name, name, name);
     }
 
-    void write_winrt_typemapgroup_assembly_attribute(writer& w, TypeDef const& type)
+    void write_winrt_comwrappers_typemapgroup_assembly_attribute(writer& w, TypeDef const& type, bool is_value_type)
     {
         auto projection_name = w.write_temp("%", bind<write_type_name>(type, typedef_name_type::NonProjected, true));
         w.write(
 R"(#pragma warning disable IL2026
 [assembly: TypeMap<WindowsRuntimeComWrappersTypeMapGroup>(
-    value: "Windows.Foundation.IReference<%>",
+    value: "%",
     target: typeof(%),
     trimTarget: typeof(%))]
 #pragma warning restore IL2026
 
-)", projection_name, projection_name, projection_name);
+)",
+        bind([&](writer& w) {
+            if (is_value_type)
+            {
+                w.write("Windows.Foundation.IReference<%>", projection_name);
+            }
+            else
+            {
+                w.write(projection_name);
+            }
+        }),
+        projection_name, 
+        projection_name);
+    }
+
+    void write_winrt_idic_typemapgroup_assembly_attribute(writer& w, TypeDef const& type)
+    {
+        // Generic interfaces are handled by cswinrtgen.
+        if (distance(type.GenericParam()) != 0)
+        {
+            return;
+        }
+
+        if (is_exclusive_to(type) && !settings.idic_exclusiveto)
+        {
+            return;
+        }
+
+        w.write(
+            R"(
+[assembly: TypeMapAssociation<DynamicInterfaceCastableImplementationTypeMapGroup>(
+    typeof(%),
+    typeof(%))]
```

Comment:
Nit: name the params just for clarity
```suggestion
    source: typeof(%),
    proxy: typeof(%))]
```